### DISCUSSION TITLE
Account restricted from Redeeming Rewards

### DISCUSSION THREAD

**Discussion by shopbuy11** [charlesbel/Microsoft-Rewards-Farmer#264] [General]
This is not a bug of the bot per se, but apparently MSFT has implemented something that makes the bot less useful now...

Until recently (back in June), I had used 5 accounts for years and farmed and redeemed quite a bit of reward points.    Then all the sudden all 5 accounts got banned completely.   So of course I just created a few new accounts. :)  Initially they worked fine - I could farm and redeem points.   Then starting a couple of weeks ago, whenever I tried to redeem, it just says that account is restricted from redeeming.   They all can farm and earn points daily without issue, but they can't redeem.  Does this happen to anyone else?   Is it happening newly created accounts only?   Any way around this?  TIA.

![image](https://github.com/charlesbel/Microsoft-Rewards-Farmer/assets/136040705/c075f5d4-b8d0-48d8-8da9-8a6ac3f668aa)

**Comment by udu3324** [comment]
This happened to me too

**Comment by tony11306** [comment]
it looks like it's not newly created account only, cuz my main account, which did run the script, got restricted as well

**Comment by udu3324** [comment]
I got banned, started getting points without the bot for 2 ish days, unbanned

**Comment by joshuamooree** [comment]
I started getting the same issue.  Not sure exactly what the cause is, but selenium is detectable by websites, so that could be at least a part of it.  There's a modified version of chromedriver that should be more stealthy.  I created a fork to use that and created a pull request: #248 .  Do note though that this adds a new dependency.  I allow the script to work without undetected_chromedriver, but you obviously don't get the stealth if you do that.

**Comment by shopbuy11** [comment]
> I started getting the same issue. Not sure exactly what the cause is, but selenium is detectable by websites, so that could be at least a part of it. There's a modified version of chromedriver that should be more stealthy. I created a fork to use that and created a pull request: #248 . Do note though that this adds a new dependency. I allow the script to work without undetected_chromedriver, but you obviously don't get the stealth if you do that.

Thanks a lot.  I replaced the main.py with your branch.  Is that enough?   Do I need to use a different version of ChromeDriver?  If so, where can I get it.

**Comment by shopbuy11** [comment]
> I got banned, started getting points without the bot for 2 ish days, unbanned

Will try that and see if it works for me.  Thanks!

**Comment by joshuamooree** [comment]
> > I started getting the same issue. Not sure exactly what the cause is, but selenium is detectable by websites, so that could be at least a part of it. There's a modified version of chromedriver that should be more stealthy. I created a fork to use that and created a pull request: #248 . Do note though that this adds a new dependency. I allow the script to work without undetected_chromedriver, but you obviously don't get the stealth if you do that.
> 
> Thanks a lot. I replaced the main.py with your branch. Is that enough? Do I need to use a different version of ChromeDriver? If so, where can I get it.

No, the files you really need are src/browser.py and src/userAgentGenerator.py.  src/browser.py is where the selenium instance gets created, so that's the heart of the change.  But undetected-webdriver is a bit more fussy with how you set user agent than normal webdriver, so I had to add some stuff to userAgentGenerator.py for that.  The new version is also a bit more stealthy because there's some headers that Chrome sends that have additional identifying info in addition to the user agent.  The new version should change those headers to be closer to what Edge uses.  I'd recommend just replacing all the .py files since there's changes to use logging instead of print statements in several of them (should still work, you just won't get some of the progress stuff logged to a text file).  To get the alternate chromedriver, just run "pip install -r requirements.txt" again (with my modified requirements.txt, of course) and it'll get pulled in automatically.  If it's all working correctly, you'll see an extra line at startup noting that the chromedriver is being patched (something undetected-chromedriver does, not me).

**Comment by shopbuy11** [comment]
Thanks again.  I shall try and report results.

**Comment by charlesbel** [comment]
Just to let everybody know I am currently writing over @joshuamooree PR to enhance even better what he done. Will soon publish it. Thanks for everybody's patience.

**Comment by joshuamooree** [comment]
> Just to let everybody know I am currently writing over @joshuamooree PR to enhance even better what he done. Will soon publish it. Thanks for everybody's patience.

One thing I would do (have been meaning to do on my fork) is check the SEC-CH-UA-* headers between desktop Edge and Mobile Edge.  I'm not 100% convinced that I have it emulating them right.  Pretty sure SEC-CH-UA is right (note that the version number for Not.A/Brand is technically supposed to be random along with the order for the items listed in that header, but I have it hard coded), but there's a bunch of other ones (SEC-CH-UA-ARCH, SEC-CH-UA-MOBILE, SEC-CH-UA-MODEL, and others) that need checking.

**Comment by tony11306** [comment]
Here to report my results. For these two or three days, I've tested account that doesn't run with the script, and like what @udu3324 said, my main account is able to redeem now. I also tested my sub account that runs the undetected version script, and it also let me redeem rewards now!(only tested on the "save the children" donation option)

**Comment by charlesbel** [comment]
Updated the script ! Please try it and give feedbacks, I'll let this issue open so we can track how things are going. Thanks @joshuamooree !

**Reply by shopbuy11** [reply]
Thanks,  you guys are great.   I will run the new version for a few days and report back if I can redeem rewards.   Is it safe to assume that the latest version incorporated all of @joshuamooree's changes?

**Reply by joshuamooree** [reply]
Yes, it appears to have all my changes and then some.

**Comment by charlesbel** [comment]
Converted this issue into a discussion so will be easier to follow.

**Comment by theskid31** [comment]
unable to use auto redeem in the xbox rewards app,I keep getting the try again later message
thanks for your hardwork,but
is this due to being detected,im sure i ran your updated version ,only started a couple of days ago

**Comment by GameSterDamian** [comment]
So is it undetected now?

**Reply by udu3324** [reply]
not sure yet

**Reply by theskid31** [reply]
So the modified chrome driver version is the one available at the mo?

**Comment by shopbuy11** [comment]
Just an update - I have been using the new undetected version for about 1 week now, minus 2 days that I couldn't get it to work with Chrome 116.  So far, my restricted accounts are still restricted, unfortunately...

Edit - I just tried to redeem for a Starbucks GC and it let me.  So I just can't redeem Amazon Gift cards for some reason.  Anyone else experiencing this or know how to fix?  Thanks!

**Reply by udu3324** [reply]
I think this happened to me too. Did the redeem page also ask you to share/confirm your location?

**Reply by shopbuy11** [reply]
No it didn't ask me.   For Amazon GC, it errors after I click "Confirm".   For Starbucks, it just shows order information and link to get the card.   In the past, when I redeemed Amazon, it asked for a cell phone to text a code to, about every other time I redeemed...

**Comment by tony11306** [comment]
I have a question is that how do you guys redeem rewards using multiple accounts? It asks me the phone number to send the code, but I only have one phone number, and I don't want to be banned from using same phone number on every account. Is there any way to bypass it or cheese it?

**Reply by udu3324** [reply]
You could get another phone number by using Google Voice

**Reply by tony11306** [reply]
That's a good choice, but sadly I can't use it in our region :(

**Reply by shopbuy11** [reply]
Actually GV numbers don't work because it rejects VOIP numbers.   Before the recent round of account bans, I did use the same cell number to redeem from 5 accounts for years without issue.  But then again I lost all of those accounts, maybe because they did catch up on that.   So I don't know...   With new accounts created lately, it won't let me redeem Amazon GCs, but does let me do Starbucks without asking for a phone.   No idea if that's going to last.   I think this party is getting shaky and likely over soon   :(
